src/coreclr/vm/threads.h

```diff
@@ -4039,9 +4039,22 @@ struct cdac_offsets<Thread>
 {
     static constexpr size_t Id = offsetof(Thread, m_ThreadId);
     static constexpr size_t OSId = offsetof(Thread, m_OSThreadId);
+    static constexpr size_t State = offsetof(Thread, m_State);
```

Comment:
Are all of the State bits part of the contract or only some of them?

Reply:
> all are mapped to a nice name for !threadstate

Except that number of the bits were reused for different purpose and the nice name is bogus for current .NET. For example, `0x40` is `TS_ExecutingOnAltStack` but SOS is going to print "Yield Requested" as its description..

> TS_Hijacked, TS_TPWorkerThread, TS_CompletionPortThread

`TS_CompletionPortThread` is never set anymore. I am deleting it in https://github.com/dotnet/runtime/pull/103394 .

We should find the bits that are actually used by the diagnostic tools for something useful (`threadstate` SOS command does not count), define those as being part of the diagnostic contract, and leave all the other bits unspecified/undefined in the diagnostic contract.

Reply:
Yes, I'd like to see us encode somewhere which bits in the implementation map to which bits in the api. We regularly muck with the bits, so just a blind copy over will eventually cause us problems.